### ISSUE TITLE
Analyzer classes cannot be created because assembly 'Microsoft.CodeAnalysis.Analyzers' is not found

### ISSUE THREAD

**Issue by cremor** [dotnet/roslyn#36570]
**Version Used**: 
The one shipped with VS2019 16.1.3

**Steps to Reproduce**:
I'm not sure how to reproduce this issue, but it is related to the analyzers that are used if you have a ASP.NET Core web project in your solution. Sometimes when I open my solution in Visual Studio the errors (see below) are there.
With VS2017 I could reproduce it quite easily even with a completely empty project (just created from the Web API template). See dotnet/roslyn-analyzers#2340 for details. But with VS2019 the error happens **far** less often, so I can't really test what exactly triggers it.

**Expected Behavior**:
No warnings should be shown in the error list of Visual Studio.

**Actual Behavior**:
The error list of Visual Studio shows these warnings:

Note: The first exception says that it searches for version 1.1.0.0 of the assembly Microsoft.CodeAnalysis.Analyzers. All other exceptions search for version 3.1.0.0.

`An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticDescriptorCreationAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

`An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.FixAnalyzers.CSharpFixerWithFixAllAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=3.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

`An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpRegisterActionAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=3.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

`An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpReportDiagnosticAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=3.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

`An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticAnalyzerFieldsAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=3.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

3x the same exception:
`Analyzer 'Microsoft.CodeAnalysis.CSharp.Analyzers.CSharpImmutableObjectMethodAnalyzer' threw an exception of type 'System.TypeInitializationException' with message 'The type initializer for 'Microsoft.CodeAnalysis.CSharp.Analyzers.CSharpImmutableObjectMethodAnalyzer' threw an exception.'.
'System.TypeInitializationException: The type initializer for 'Microsoft.CodeAnalysis.CSharp.Analyzers.CSharpImmutableObjectMethodAnalyzer' threw an exception. ---> System.IO.FileNotFoundException: Die Datei oder Assembly "Microsoft.CodeAnalysis.Analyzers, Version=3.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.`

My initial report of this issue for VS2017: dotnet/roslyn-analyzers#2340
Maybe related: #32226

**Comment by mavasani** [comment]
I am unable to repro this with the latest VS2019 16.2 preview4 bits. I think this is a duplicate of https://github.com/dotnet/roslyn/issues/32226

**Comment by cremor** [comment]
> I think this is a duplicate of #32226

If it is, why can I still reproduce it with VS2019 16.1.3? According to the milestone in that bug, it was fixed in a 16.0 preview.

**Comment by mavasani** [comment]
@cremor Do you have a consistent repro? Or it just repros intermittently on creating a project?

**Comment by cremor** [comment]
No, I don't have a consistent repro. I rarely (**very** rarely) get it when I open my big solution that contains a ASP.NET Core web project.

I think I've never seen it right after creating a project - only after opening an existing one.

It was easier to reproduce in VS2017 - there I had a chance of about 1:5 to get it. But with VS2019 it is very hard to reproduce. I think I got it only 2 or 3 times since I updated to VS2019 (about a month ago), while I opened the solution countless times in that time span...

Sorry that that's not very helpful 😢

**Comment by jinujoseph** [comment]
Closing this issue, feel free to reach back if you hit this again and is reproducible

**Comment by IanKemp** [comment]
I am seeing this a lot in VS2019 16.3.4 with a solution containing a .NET Core 2.2 ASP.NET web project.

```
An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticDescriptorCreationAnalyzer cannot be created from C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Could not load file or assembly 'Microsoft.CodeAnalysis.Analyzers, Version=3.3.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
```

**Comment by goforgold** [comment]
I am seeing the same issue in VS 2017 15.9.24

**Comment by giantpenguin** [comment]
I would like provide what I have done to resolve the similar issue, un-check the Analyzers on the Code Analysis of the Properties page of the project has this issue:

![image](https://user-images.githubusercontent.com/7245621/88646832-a3ace680-d0ef-11ea-8bb8-76378c0f865f.png)

**Comment by IanKemp** [comment]
This issue still persists in VS 2021 (latest 17.1.0).

> I would like provide what I have done to resolve the similar issue, un-check the Analyzers on the Code Analysis of the Properties page of the project has this issue:
> 
> ![image](https://user-images.githubusercontent.com/7245621/88646832-a3ace680-d0ef-11ea-8bb8-76378c0f865f.png)

We're asking what to do after we've been shot in the foot, and you're telling us to not get shot in the foot. Thanks...

**Comment by CyrusNajmabadi** [comment]
@IanKemp can you provide us with your solution and the steps you took to repro this?

**Comment by IanKemp** [comment]
> @IanKemp can you provide us with your solution and the steps you took to repro this?

Sadly not, as it's my employer's sln :/. And the steps are literally "build the sln". Are there any logs I can collect to help? I could scrub them enough to make them non-identifiable.

**Comment by CyrusNajmabadi** [comment]
@mavasani @jmarolf @jasonmalinowski are there any logs that Ian can collect for us here?

**Comment by Tracy-P** [comment]
This is the closest issue to my problem. I hope this is something that can be tracked down.
I have the same solution I work on that has 15 projects in it. I am transitioning to a new computer and this happens on both.
The solution originally started as .NET Framework and has been updated to .NET Core 3.1 at some point. 
Both CheckInReports and PcKiosk target netcoreapp3.1.
PcKiosk uses both WPF and WinForms.
I am not in a position that I can share the solution.

Visual Studio 2022 v17.4.0

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.FixAnalyzers.CSharpFixerWithFixAllAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Could not load type 'Microsoft.CodeAnalysis.Analyzers.FixAnalyzers.FixerWithFixAllAnalyzer`1' from assembly 'Microsoft.CodeAnalysis.Analyzers, Version=2.9.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'..	CheckInReports	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticDescriptorCreationAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Could not load type 'Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.DiagnosticDescriptorCreationAnalyzer`3' from assembly 'Microsoft.CodeAnalysis.Analyzers, Version=2.9.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'..	CheckInReports	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpRegisterActionAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Method 'GetCodeBlockAnalyzer' in type 'Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpRegisterActionAnalyzer' from assembly 'Microsoft.CodeAnalysis.CSharp.Analyzers, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation..	CheckInReports	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.FixAnalyzers.CSharpFixerWithFixAllAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Could not load type 'Microsoft.CodeAnalysis.Analyzers.FixAnalyzers.FixerWithFixAllAnalyzer`1' from assembly 'Microsoft.CodeAnalysis.Analyzers, Version=2.9.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'..	PcKiosk	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticDescriptorCreationAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Could not load type 'Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.DiagnosticDescriptorCreationAnalyzer`3' from assembly 'Microsoft.CodeAnalysis.Analyzers, Version=2.9.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'..	PcKiosk	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active
Warning	CS8032	An instance of analyzer Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpRegisterActionAnalyzer cannot be created from C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll: Method 'GetCodeBlockAnalyzer' in type 'Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpRegisterActionAnalyzer' from assembly 'Microsoft.CodeAnalysis.CSharp.Analyzers, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' does not have an implementation..	PcKiosk	C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll	1	Active

```

Only these two projects are showing the warnings.

Here's some things I'm seeing:
![image](https://user-images.githubusercontent.com/47571263/201226912-86e61864-3f22-42ee-a969-6f9285292d22.png)

In the smallest project that has 1 nuget package installed. (CheckInReports)
When I go to update the nuget packages I see 207 transitive packages. I don't know what that means, I don't need to right now. I sure would like to send you that list.
In that transitive list I find this:
![image](https://user-images.githubusercontent.com/47571263/201227079-2b864bdd-d0f7-497f-ab9e-9726077a7500.png)

❗**If I upgrade that transitive 1.1.0 package to 3.3.3, the warnings go away. But this project doesn't need this, so I uninstall it. The warnings come back.**

Without it installed the csproj has the following:
```
  <ItemGroup>
    <PackageReference Include="DevExpress.AspNetCore.Reporting" Version="22.1.4" />
  </ItemGroup>
```

The PcKiosk project does not have the transitive nuget package that the CheckInReports project does.

I installed the 1.1.0 on the new machine and the problem still happens. 
Even with 3.3.3 still in the .nuget/packages folder the warning continues to be here.

For the CheckInReports project.
I deleted the bin and obj folders and did a rebuild. The obj\CheckInReports.csproj.nuget.g.props contains:
```
  <PropertyGroup Condition=" '$(ExcludeRestorePackageImports)' != 'true' ">
    <PkgNewtonsoft_Json Condition=" '$(PkgNewtonsoft_Json)' == '' ">C:\Users\tracy.POWERCHURCH\.nuget\packages\newtonsoft.json\10.0.1</PkgNewtonsoft_Json>
    <PkgMicrosoft_CodeAnalysis_Analyzers Condition=" '$(PkgMicrosoft_CodeAnalysis_Analyzers)' == '' ">C:\Users\tracy.POWERCHURCH\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0</PkgMicrosoft_CodeAnalysis_Analyzers>
  </PropertyGroup>
````

❓ **Is there any place to find out how that Transitive list is built? **

In a third project that also has the DevExpress.AspNetCoreReporting nuget package I see this in the Transitive package list. **NOTE:** This project does not have warnings.
![image](https://user-images.githubusercontent.com/47571263/201232049-cb0d6673-7d90-4386-bb32-d46a5e5b618b.png)

2.9.4 is also installed on my new PC.

To throw another twist into this. A third machine with the same project, but no locally installed DevExpress shows the warnings, but has no Transitive packages listed. (red herring?)
